sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -1415,6 +1415,20 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     )
   }
 
+  test("SPARK-28386: Cannot resolve ORDER BY columns with GROUP BY and HAVING") {
+    assertAnalysisSuccess(
+      parsePlan(
+        """
+          |WITH t1 as (SELECT 1 id, 'one' name)
```

Comment:
we can put this in golden file test as well.

Reply:
Since this just checks analysis success, we can remove it if the test is already in golden file.